diracos/diracoslib.py

```diff
@@ -482,8 +482,19 @@ def buildPackage(packageCfg):
 
     _root, srcExtension = os.path.splitext(src)
 
-    if srcExtension == '.rpm':  # In fact, it is .src.rpm
-      _buildFromSRPM(packageCfg)
+    if srcExtension == '.rpm':
+      if src.endswith('.src.rpm'):  # In fact, it is .src.rpm
+        _buildFromSRPM(packageCfg)
+      else:  # prebuilt RPM
+        repository = packageCfg['repo']
+        rpmUrl = packageCfg['src']
+        buildOnly = packageCfg.get('buildOnly', False)
+        rpmDestDir = repository
+        # We only build, so we copy the rpms to a special directory
+        if buildOnly:
```

Comment:
but otherwise it ends up in the root directory of the yum repo, not in a subdirectory ?

Reply:
Well spotted! As there is no use case yet, I've made it raise a `NotImplementedError`  for the other branch to ensure everything in the final tarball has been built from source.